frontends/web/src/components/backbutton/backbutton.tsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2024 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactNode, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useEsc } from '@/hooks/keyboard';
+import { Button } from '@/components/forms/button';
+
+type TBackButton = {
+  children: ReactNode;
+  className?: string;
+  disabled?: boolean;
+  enableEsc?: boolean;
+};
+
+export const BackButton = ({
+  children,
+  className,
+  disabled,
+  enableEsc,
+}: TBackButton) => {
+  const navigate = useNavigate();
+
+  const handleBack = useCallback(() => {
+    if (!disabled) {
+      navigate(-1);
+    }
+  }, [disabled, navigate]);
+
+  useEsc(() => enableEsc && handleBack());
```

Comment:
i am never quite sure how it works, but maybe you need useCallback(() => ...)) here to avoid re-renders?

Reply:
I tested a bit and found a perf issue, but the other way around. 

If there are many re-renders this causes the internal useKeyboard hook to remove and re-attach the keyboad event listener on document.

On transaction overview this re-attached the event over 100 times (1 per tx). Added a commit to address this and not render Dialog when there is no data.